lib/reda/importers/mpt_das1.py

```diff
@@ -0,0 +1,621 @@
+
+    Parameters
+    ----------
+    filename: string
+        input filename
+    header_row: int
+        row number of header row
+
+    Returns
+    -------
+    frequencies: list
+        Contains the measured frequencies
+
+    Notes
+    -----
+    .
+    """
+
+    fid = open(filename, 'r')
+    lines = fid.readlines()
+    freq_header = lines[header_row+1].split('Hz')[:-1]
+    frequencies = [re.sub("[^\d\.]", "", part) for part in freq_header]
+    fid.close()
+    return frequencies
+
+
+def import_das1_fd(filename, **kwargs):
+    """Reads a frequency domain (single frequency) MPT DAS-1 data file (.Data)
+    and prepares information in pandas DataFrame for further processing.
+
+    Parameters
+    ----------
+    filename : string
+        path to input file
+    corr_array : list, optional
+        correction values for electrode numbers [a, b, m, n]
+
+    Returns
+    -------
+    data : :py:class:`pandas.DataFrame`
+        Contains the measurement data
+    electrodes : :py:class:`pandas.DataFrame`
+        Contains electrode positions (None at the moment)
+    topography : None
+        No topography information is contained in the text files, so we always
+        return None
+
+
+    """
+    if 'corr_array' in kwargs:
+        corr_array = kwargs.get('corr_array')
+    else:
+        corr_array = [0, 0, 0, 0]
+
+    print(corr_array)
+
+    df = pd.read_csv(filename,
+                     delimiter=' ',
+                     comment='!',
+                     index_col=0)
+
+    # derive rows used in data block
+    data_start = df.index.get_loc('#data_start')
+    data_end = df.index.get_loc('#data_end')
+    data = df.iloc[data_start+1: data_end].dropna(axis=1)
+
+    data_new = pd.DataFrame()
+    # A, B, M, N
+    data_new['a'] = [int(x.split(',')[1])-corr_array[0] for x in data.iloc[:, 0]]
+    data_new['b'] = [int(x.split(',')[1])-corr_array[1] for x in data.iloc[:, 1]]
+    data_new['m'] = [int(x.split(',')[1])-corr_array[2] for x in data.iloc[:, 2]]
+    data_new['n'] = [int(x.split(',')[1])-corr_array[3] for x in data.iloc[:, 3]]
+
+    data_new['r'] = np.array(data.iloc[:, 4]).astype('float')  # resistance
+    data_new['rpha'] = np.array(data.iloc[:, 5]).astype('float')  # phase
+    data_new['I'] = np.array(data.iloc[:, 12]).astype('float')  # current in mA
+    data_new['Zt'] = data_new['r'] * np.exp(data_new['rpha'] * 1j / 1000.0)
+
+    datetime_series = pd.to_datetime(data.iloc[:, -7],
+                                     format='%Y%m%d_%H%M%S',
+                                     errors='ignore')
+
+    data_new['datetime'] = [time for index, time in datetime_series.iteritems()]
+
+    return data_new, None, None
+
+
+def import_das1_td(filename, **kwargs):
+    """Reads a time domain MPT DAS-1 data file (.Data) and prepares information
+    in pandas DataFrame for further processing.
+
+    Parameters
+    ----------
+    filename : string
+        path to input file
+    corr_array : list, optional
```

Comment:
Could you provide an explanation (and possibly a short use-case/example) for corr_array? This seems to be some correction for cases where electrode positions were not put in correctly, similar to common practice for the Syscal... ?

Reply:
This is meant to correct electrode numbers for layouts with separated cables for current and potential measurements, where you have to break down the electrode numbering to 32 electrodes. I see that it's not really necessary here and could also be done in the processing afterwards. Hence, I'm fine with removing it but leave the decision to you.